PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx

```diff
@@ -0,0 +1,700 @@
+  Configurable<float> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<float> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+
+  HfHelper hfHelper;
+  SliceCache cache;
+
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelLc>> selectedLcCandidates = aod::hf_sel_candidate_lc::isSelLcToPKPi >= selectionFlagLc || aod::hf_sel_candidate_lc::isSelLcToPiKP >= selectionFlagLc;
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelLc, aod::HfCand3ProngMcRec>> selectedLcCandidatesMc = aod::hf_sel_candidate_lc::isSelLcToPKPi >= selectionFlagLc || aod::hf_sel_candidate_lc::isSelLcToPiKP >= selectionFlagLc;
+
+  void processLcSelectionData(aod::Collision const& collision,
+                              soa::Join<aod::HfCand3Prong, aod::HfSelLc> const& candidates)
+  {
+    bool isLcFound = 0;
+    if (selectedLcCandidates.size() > 0) {
+      auto selectedLcCandidatesGrouped = selectedLcCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex(), cache);
+
+      for (const auto& candidate : selectedLcCandidatesGrouped) {
+        if (!TESTBIT(candidate.hfflag(), aod::hf_cand_3prong::DecayType::LcToPKPi)) {
+          continue;
+        }
+        if (yCandMax >= 0. && std::abs(hfHelper.yLc(candidate)) > yCandMax) {
+          continue;
+        }
+        if (ptCandMin >= 0. && candidate.pt() < ptCandMin) {
+          continue;
+        }
+        isLcFound = 1;
+        break;
+      }
+    }
+    lcSel(isLcFound);
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadronsSelection, processLcSelectionData, "Process Lc Collision Selection Data", true);
+
+  void processLcSelectionMcRec(aod::Collision const& collision,
+                               soa::Join<aod::HfCand3Prong, aod::HfSelLc, aod::HfCand3ProngMcRec> const& candidates)
+  {
+    bool isLcFound = 0;
+    if (selectedLcCandidatesMc.size() > 0) {
+      auto selectedLcCandidatesGroupedMc = selectedLcCandidatesMc->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex(), cache);
+      for (const auto& candidate : selectedLcCandidatesGroupedMc) {
+        // check decay channel flag for candidate
+        if (!TESTBIT(candidate.hfflag(), aod::hf_cand_3prong::DecayType::LcToPKPi)) {
+          continue;
+        }
+        if (yCandMax >= 0. && std::abs(hfHelper.yLc(candidate)) > yCandMax) {
+          continue;
+        }
+        if (ptCandMin >= 0. && candidate.pt() < ptCandMin) {
+          continue;
+        }
+        isLcFound = 1;
+        break;
+      }
+    }
+    lcSel(isLcFound);
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadronsSelection, processLcSelectionMcRec, "Process Lc Selection McRec", false);
+
+  void processLcSelectionMcGen(aod::McCollision const& mcCollision,
+                               aod::McParticles const& mcParticles)
+  {
+    bool isLcFound = 0;
+    for (const auto& particle : mcParticles) {
+      if (std::abs(particle.pdgCode()) != pdg::Code::kLambdaCPlus) {
+        continue;
+      }
```

Comment:
Is it intended to check only the PDG code and not the decay channel?

Reply:
Ciao Fabrizio, yes, that selection is just used to understand which events has to be considered for the mixed-event, and there it's enough to have Lc particle in the event, not necessary that it decays in the specific decay channel used in the analysis

Reply:
@fcolamar thanks for the reply! Wouldn't it be more efficient to have 2 tables that one can fill in one go?

Reply:
Yes that alternate approach would also work, but the original idea was to avoid having a proliferation of tables, also because the McGen is expected to be employed only for the MC closure check, so to be seldom used